src/MyTested.AspNetCore.Mvc.Controllers.Attributes/Builders/Attributes/AuthorizeAttributeTestBuilder.cs

```diff
@@ -0,0 +1,142 @@
+        public IAndAuthorizeAttributeTestBuilder WithPolicy(string policy)
+        {
+            if (!string.IsNullOrEmpty(policy))
+            {
+                this.Attribute = new AuthorizeAttribute(policy)
+                {
+                    Roles = this.Attribute.Roles,
+                    AuthenticationSchemes = this.Attribute.AuthenticationSchemes
+                };
+
+                this.Validations.Add((expected, actual) =>
+                {
+                    var expectedPolicy = expected.Policy;
+                    var actualPolicy = actual.Policy;
+
+                    if (!string.Equals(expectedPolicy, actualPolicy, StringComparison.OrdinalIgnoreCase))
+                    {
+                        this.FailedValidationAction(
+                            $"{this.ExceptionMessagePrefix}'{expectedPolicy}' policy",
+                            $"in fact found '{actualPolicy}'");
+                    }
+                });
+            }
+
+            return this;
+        }
+
+        /// <inheritdoc />
+        public IAndAuthorizeAttributeTestBuilder WithRole(string role)
+        {
+            if (!string.IsNullOrEmpty(role))
+            {
+                this.Attribute.Roles = role;
+
+                this.Validations.Add((expected, actual) =>
+                {
+                    var expectedRoles = expected.Roles;
+                    var actualRoles = actual.Roles;
+
+                    if (!string.Equals(expectedRoles, actualRoles, StringComparison.OrdinalIgnoreCase))
+                    {
+                        this.FailedValidationAction(
+                            $"{this.ExceptionMessagePrefix}'{expectedRoles}' roles",
+                            $"in fact found '{actualRoles}'");
+                    }
+                });
+            }
+
+            return this;
+        }
+
+        /// <inheritdoc />
+        public IAndAuthorizeAttributeTestBuilder WithRoles(IEnumerable<string> roles)
+        {
+            if (roles != null && roles.Any())
+            {
+                this.Attribute.Roles = string.Join(",", roles);
+
+                this.Validations.Add((expected, actual) =>
+                {
+                    var expectedRoles = expected.Roles;
+                    var actualRoles = actual.Roles;
+
+                    if (!string.Equals(expectedRoles, actualRoles, StringComparison.OrdinalIgnoreCase))
+                    {
+                        this.FailedValidationAction(
+                            $"{this.ExceptionMessagePrefix}'{expectedRoles}' roles",
+                            $"in fact found '{actualRoles}'");
+                    }
+                });
+            }
+
+            return this;
+        }
+
+        /// <inheritdoc />
+        public IAndAuthorizeAttributeTestBuilder WithRoles(params string[] roles)
+        {
+            if (roles != null && roles.Any())
+            {
+                return this.WithRoles(new List<string>(roles));
```

Comment:
Use `this.WithRoles(roles.AsEnumerable())` because that is the convention in the project.